lib/ddtrace/tracer.rb

```diff
@@ -141,6 +141,8 @@ def trace(name, options = {})
     # Record the given finished span in the +spans+ list. When a +span+ is recorded, it will be sent
     # to the Datadog trace agent as soon as the trace is finished.
     def record(span)
+      span.service ||= Datadog::Span::DEFAULT_SERVICE
```

Comment:
Use case 1:
* I create a brand new trace with:
```
require 'ddtrace'
Datadog.tracer.trace("foo")
```
* now this works, before we got a drop because the service was "optional" but in real it's required

Use case 2:
* I have a Rails automatic integration
* for some reasons (bugs), it drops the root span suddenly so inner spans will not inherit the ``parent.service``
* they're reported as a new ``ruby`` service instead of being dropped

Not sure if the use case 2 is a good thing (will follow with more details).